### ISSUE TITLE
Add an option to change how folder names are generated when a new share is accepted

### ISSUE THREAD

**Issue by passcod** [syncthing/syncthing#6879]
Currently when a new share is (automatically) accepted the generated name on disk is the "human" name of the share, falling back to the ID if that exists on disk already. I'd like an option to always use the ID (and possibly inverse the fallback then? not sure if that would be useful).

### Specific use case

I have a central server that stores and backs up shares of the documents, desktop, etc folders on the various machines I use. The shares are not sent to other devices, it's a 1:1 device:server thing. For convenience, on the devices the shares would be created as "Documents" with an ID of `documents/devicename`, and on the server I'd rename them to "Documents / DeviceName".

### Workarounds

Currently I have to create them on device as "documents/devicename" with the same ID, wait for sync to establish so the folder is created on the server, and then rename them both on device and on server. That's tedious and error-prone; to fix a mistake I then have to SSH to the server.

Alternatively I could make `Documents` etc folders on the server so the names would conflict and the fallback to ID would trigger, but that seems even more brittle (relying on a fallback behaviour) and doesn't scale if a new share "category" is added.

### General

As the ID is permanent, and the friendly name changeable, it seems to me more consistent/less surprising to use the folder ID when auto-accepting shares. However of course changing it outright would not be backward compatible, and I can see that defaulting to the friendly name is better in the interactive case. Thus: an option.

**Comment by calmh** [comment]
Yeah... much as I dislike options for simple behavior the use case seems to make sense. For most users the current behavior is desirable because when using the GUI to add a folder you'll end up with an auto generated folder ID like `fjguk-vriqn` but probably set a label like `Photos 2020` and the latter is generally more suitable as the on disk directory name.

Having a checkbox to ignore the label and always use folder ID seems reasonable.

**Comment by AudriusButkevicius** [comment]
It already uses the ID if there is an existing folder with the same label.

**Comment by calmh** [comment]
Yeah but as noted that depends on having a pre-created folder to conflict with the other side's label, and that's not very safe or predictable.

**Comment by calmh** [comment]
One could imagine a template for default folder paths. Currently it's always `~/$label` but you could imagine being able to set it to `/folders/auto/$id` or even `/folders/auto/$id-$label` or such.

**Comment by brunosxs** [comment]
bumping this as it is very disruptive the normal behavior.  I add some folders and give them descriptive names that reflect the relative structure... and it sucks when I have auto accept shares because it completely destroys expectation.
I share:
```
work/projecta
hobbies/c/gbemu
hobbies/c/3drenderer
```
and I expect that to be preserved on the sync folder on the other computer, but instead I get everything inside the sync folder as:
```
work projecta
hobbies c gbemu
hobbies c 3drenderer
```
I think of this at best as a lack of a feature and at worst as a bug.

**Comment by acolomb** [comment]
@lietu can you please take this to our forum at https://forum.syncthing.net/?

It's not at all clear what you are trying to do and what happened.  My gut feeling is that you shouldn't be using the auto-accept feature if you want control over the paths of newly added folders.  But to clarify, this requires more conversation than we want to spam the issue tracker with.

**Comment by olfway** [comment]
> One could imagine a template for default folder paths. Currently it's always `~/$label` but you could imagine being able to set it to `/folders/auto/$id` or even `/folders/auto/$id-$label` or such.

This would be perfect for a central syncthing on server with multiple devices connected

Labels could be non-equal and as a result some folder will be "~/$label" and second one will be "~/$name" which is non-consistent
